### PR TITLE
fix: integration tests

### PR DESCRIPTION
**Pull request by JadhavPoonam**
### Description

Use correct docker images in integration tests

### PR Checklist

* [ ] ~updated test coverage~
* [ ] ~external facing docs updated~
* [x] appropriate backport labels added
* [ ] ~not a security concern~

### CODE REVIEW

File: test/integration/consul-container/libs/utils/version.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+func init() {
+	flag.StringVar(&TargetImage, "target-image", defaultImageName, "docker image name to be used under test (Default: "+defaultImageName+")")
+	flag.StringVar(&TargetVersion, "target-version", "local", "docker image version to be used as UUT (unit under test)")
+
+	flag.StringVar(&LatestImage, "latest-image", defaultImageName, "docker image name to be used under test (Default: "+defaultImageName+")")
+	flag.StringVar(&LatestVersion, "latest-version", "latest", "docker image to be used as latest")
+}
+
+func DockerImage(image, version string) string {
```
Comment (dhiaayachi) [inline test/integration/consul-container/libs/utils/version.go:31]: Why not have 2 versions of this func:
- 1 in OSS that return empty suffix
- 1 in ENT that return "-ent" suffix

**Review by dhiaayachi — COMMENTED** [review]
Hey @JadhavPoonam, I really like the idea in here, I left few comments.

File: test/integration/consul-container/libs/utils/version.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package utils
```
Comment (asheshvidyut) [inline test/integration/consul-container/libs/utils/version.go:1]: we can add 
```
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0
```

File: test/integration/consul-container/libs/utils/version_oss.go [hunk 3]
```diff
@@ -3,12 +3,7 @@
...
-// TODO: need a better way to abstract the container creation and configuration;
-//       please refer to the discussion in github PR
-
-var TargetImage = flag.String("target-image", "consul", "docker image name to be used under test (Default: consul)")
-var TargetVersion = flag.String("target-version", "local", "docker image version to be used as UUT (unit under test)")
-var LatestImage = flag.String("latest-image", "consul", "docker image name to be used under test (Default: consul)")
-var LatestVersion = flag.String("latest-version", "1.11", "docker image to be used as latest")
+const (
+	defaultImageName   = DefaultImageNameOSS
+	ImageVersionSuffix = ""
```
[the remark this answers is not in the document]
Comment (JadhavPoonam) [reply test/integration/consul-container/libs/utils/version_oss.go:8]: Yes, this is not needed. I will clean it up

File: test/integration/consul-container/libs/service/assets/Dockerfile-consul-envoy [hunk 4]
```diff
@@ -1,7 +1,8 @@
 # Note this arg has to be before the first FROM
+ARG CONSUL_IMAGE
```
[the remark this answers is not in the document]
Comment (JadhavPoonam) [reply test/integration/consul-container/libs/service/assets/Dockerfile-consul-envoy:2]: I don't think so. AFAICT the make-file depends on the docker-file under the `build-support` directory which should  in theory use the image with `consul:local` tag

File: test/integration/consul-container/libs/utils/version.go (hunk 2 above)
Comment (JadhavPoonam) [reply test/integration/consul-container/libs/utils/version.go:1]: This PR is for the`1.14` version which does not have these headers

File: test/integration/consul-container/libs/service/assets/Dockerfile-consul-envoy (hunk 4 above)
[the remark this answers is not in the document]
Comment (dhiaayachi) [reply test/integration/consul-container/libs/service/assets/Dockerfile-consul-envoy:2]: oh ok, this one is internal to the library 👍

**Pull request merged**